### PR TITLE
feature: add editorFont setting in user preference.

### PR DESCRIPTION
**Pull request by Anderson-Liu**
| Q                | A
| ---------------- | ---
| Bug fix?         | no
| New feature?     | yes
| BC breaks?       | no
| Deprecations?    | no
| New tests added? | yes/not needed
| Fixed tickets    | 
| License          | MIT

### Description

Add editorFont setting in user preference. User can config it to custom their editor's font family.

--

#### Please, don't submit `/dist` files with your PR!

### CODE REVIEW

File: src/renderer/store/editor.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ const state = {
   },
   // user preference
   theme: 'light',
+  editorFontFamily: 'Monaco',
```
Comment (fxha) [inline src/renderer/store/editor.js]: Can you please change default font family to `Open Sans`.

File: static/preference.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ Edit and save to update preferences.  You can only change the JSON below!
 ```json
 {
   "fontSize": "16px",
+  "editorFontFamily": "Monaco",
```
Comment (fxha) [inline static/preference.md]: Can you please change default font family to `Open Sans`.

File: src/editor/index.js [hunk 3]
```diff
@@ -493,9 +493,10 @@ class Aganippe {
     this.contentState.render()
   }
 
-  setFont ({ fontSize, lineHeight }) {
+  setFont ({ fontSize, lineHeight, editorFontFamily }) {
     if (fontSize) this.fontSize = parseInt(fontSize, 10)
     if (lineHeight) this.lineHeight = lineHeight
+    if (editorFontFamily) this.editorFontFamily = editorFontFamily
```
Comment (Jocs) [inline src/editor/index.js]: is editorFontFamily used in `editor` folder? If not, why do you add it?

**Comment by Jocs** [comment]
@peacesky The `font family` is extremely unwieldy, and users may type in errors. At this point we may need to do some verification. If Chrome doesn't support or misspells, we all need a fallback mechanism.

File: src/renderer/store/editor.js (hunk 1 above)
Comment (peacesky) [reply src/renderer/store/editor.js]: Ok

File: static/preference.md (hunk 2 above)
Comment (peacesky) [reply static/preference.md]: Done!

File: src/editor/index.js (hunk 3 above)
Comment (peacesky) [reply src/editor/index.js]: Sorry, I have remove it.

**Comment by peacesky** [comment]
@Jocs Is it a good way that auto append generic family name after receive user input in `preference.js`?

File: src/renderer/components/editor.vue [hunk 4]
```diff
@@ -2,7 +2,8 @@
   <div
     class="editor-wrapper"
     :class="[{ 'typewriter': typewriter, 'focus': focus, 'source': sourceCode }, theme]"
-    :style="{ 'color': theme === 'dark' ? darkColor : lightColor, 'lineHeight': lineHeight, 'fontSize': fontSize, 'font-family': editorFontFamily }"
+    :style="{ 'color': theme === 'dark' ? darkColor : lightColor, 'lineHeight': lineHeight, 'fontSize': fontSize, 
+    'font-family': editorFontFamily ? `${editorFontFamily}, sans-serif`:'sans-serif'}"
```
Comment (Jocs) [inline src/renderer/components/editor.vue]: I think you need to judge weather `editorFontFamily ` is support be Chrome in Electron? not just judge `editorFontFamily ?`

Comment (peacesky) [reply src/renderer/components/editor.vue]: Yeap! I know that. But after refer other project like VSCode and google searching, I still can't find a good way to do that.
Something confuse me like where can I get the supported and unsupported list in electron(base on Chromium).
Is all what I need to do is check if user input's font-family in [Google font directory](https://fonts.google.com/)?
Or I have to write some tricks code to check is it work on chrome? Like use [CSS.supports()](https://developer.mozilla.org/en-US/docs/Web/API/CSS/supports) or something else?
Here you can check how VSCode(which also develop base electron)do about this: 
[Related test code](https://github.com/Microsoft/vscode/blob/165686e333508bc838d580b3738292d7fa9b82d0/src/vs/workbench/parts/terminal/test/electron-browser/terminalConfigHelper.test.ts#L36)
[Related code]
(https://github.com/Microsoft/vscode/blob/17ed2280597e68fefa00b43c8d28d0442891ea22/src/vs/workbench/parts/terminal/electron-browser/terminalConfigHelper.ts#L91)
I still can't figure out how they do something like misspells check or font-family support check in electron(base on Chromium) , and what is the better way to do that.
Please give some help! Thanks!

Comment (Jocs) [reply src/renderer/components/editor.vue]: Do you think that `fallback` font family is enough?

File: src/renderer/components/editor.vue [hunk 5]
```diff
@@ -136,6 +138,12 @@
           editor.setFont({ fontSize: value })
         }
       },
+      editorFontFamily: function (value, oldValue) {
```
Comment (Jocs) [inline src/renderer/components/editor.vue]: maybe not needed?

File: src/renderer/components/editor.vue (hunk 4 above)
Comment (Anderson-Liu) [reply src/renderer/components/editor.vue]: Is there any best best practices to implement that ?
I am also take it importance but have no idea about how to implement it in proper way.

**Comment by Jocs** [comment]
@Anderson-Liu I think we can leave it to later optimization.

**Pull request merged**